x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/RegionMapChart/ChoroplethMap.tsx

```diff
@@ -0,0 +1,269 @@
+  const enableScrollZoom = useRef(false);
+  const [hoverState, setHoverState] = useState<{
+    geojsonProperties?: GeoJsonProperties;
+    data?: ChoroplethDataElement;
+  }>({});
+  // props stored in useRefs, used in the initialization side effect
+  const initialGeojsonSourceRef = useRef(initialGeojsonSource);
+  const initialMapboxOptionsRef = useRef(initialMapboxOptions);
+
+  // converts input data to a scaled value between 0 and 1
+  const getValueScale = useCallback(
+    (value: number) => {
+      const [min, max] = getDataRange(data);
+      return (value - min) / (max - min);
+    },
+    [data]
+  );
+
+  const controlScrollZoomOnWheel = useCallback((event: WheelEvent) => {
+    if (event.ctrlKey || event.metaKey) {
+      event.preventDefault();
+    } else {
+      event.stopPropagation();
+    }
+  }, []);
+
+  const updateHoverStateOnMousemove = useCallback(
+    (event: mapboxgl.MapMouseEvent & mapboxgl.EventData) => {
+      if (map && popupRef.current && data.length) {
+        popupRef.current.setLngLat(event.lngLat);
+        const hoverFeatures = map.queryRenderedFeatures(event.point, {
+          layers: [CHOROPLETH_LAYER_ID]
+        });
+        if (hoverFeatures[0]) {
+          const geojsonProperties = hoverFeatures[0].properties;
+          // only set state when necessary since event fires so often
+          if (!isEqual(geojsonProperties, hoverState.geojsonProperties)) {
+            const matchedData = data.find(
+              ({ key }) =>
+                geojsonProperties &&
+                key === geojsonProperties[geojsonKeyProperty]
+            );
+            setHoverState({ geojsonProperties, data: matchedData });
+          }
+        } else {
+          setHoverState({});
+        }
+      }
+    },
+    [map, data, hoverState.geojsonProperties, geojsonKeyProperty]
+  );
+
+  const updateHoverStateOnMouseout = useCallback(
+    (event: mapboxgl.MapMouseEvent & mapboxgl.EventData) => {
+      enableScrollZoom.current = false;
+      setHoverState({});
+    },
+    []
+  );
+
+  // When new, memoized handlers are created with useCallback,
+  // these references allow the old handlers to be removed
+  const updateHoverStateOnMousemoveRef = useRef<
+    ((event: mapboxgl.MapMouseEvent & mapboxgl.EventData) => void) | null
+  >(null);
+
+  const updateHoverStateOnMouseoutRef = useRef<
+    ((event: mapboxgl.MapMouseEvent & mapboxgl.EventData) => void) | null
+  >(null);
+
+  // initialization side effect, only runs once
+  useEffect(() => {
+    if (containerRef.current !== null) {
+      // set up Map object
+      const mapboxMap = new Map({
+        attributionControl: false,
+        container: containerRef.current,
+        style:
+          'https://tiles.maps.elastic.co/styles/osm-bright-desaturated/style.json',
+        ...initialMapboxOptionsRef.current
+      });
+      mapboxMap.dragRotate.disable();
+      mapboxMap.touchZoomRotate.disableRotation();
```

Comment:
Do you need to disable these after initialization? Can't you set them to `false` as part of the config above?

```js
const mapboxMap = new Map({
  dragRotate: false,
  touchZoomRotate: false
})